pkg/datadog/go.mod

```diff
@@ -270,3 +296,6 @@ replace github.com/open-telemetry/opentelemetry-collector-contrib/internal/aws/e
 
 // openshift removed all tags from their repo, use the pseudoversion from the release-3.9 branch HEAD
 replace github.com/openshift/api v3.9.0+incompatible => github.com/openshift/api v0.0.0-20180801171038-322a19404e37
+
+// Replace old monolithic genproto with newer version to avoid ambiguous import with split modules
+replace google.golang.org/genproto => google.golang.org/genproto v0.0.0-20231030173426-d783a09b4405
```

Comment:
We should not do this, this only pins the dependency on this module but not on any modules that import this one, which means any customer using Datadog components on a custom build may have to add this replace to their Collector builder configuraiton

From https://go.dev/ref/mod#go-mod-file-replace:
> replace directives only apply in the main module’s go.mod file and are ignored in other modules